sdk/core/core/src/az_json_data.c

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_data.h>
+
+#include <az_json_parser.h>
+#include <az_span_builder.h>
+
+#include <_az_cfg.h>
+
+typedef struct {
+  az_span span;
+  size_t align;
+} az_data;
+
+#define AZ_DATA(V) \
+  ((az_data){ \
+      .span = (az_span){ .begin = (uint8_t const *)&(V), .size = sizeof(V) }, \
+      .align = AZ_ALIGNOF(V), \
+  })
+
+typedef struct {
```

Comment:
I'd like to see some documentation comments about how all these types relate to each other.